test/e2e/kubectl.go

```diff
@@ -652,10 +652,15 @@ var _ = framework.KubeDescribe("Kubectl client", func() {
 				err := wait.Poll(framework.Poll, timeout, func() (bool, error) {
 					endpoints, err := c.Endpoints(ns).Get(name)
 					if err != nil {
-						if apierrs.IsNotFound(err) {
+						// log the real error
+						framework.Logf("Get endpoints failed (interval %v): %v", framework.Poll, err)
+
+						// if the error is API not found or could not find default credentials or TLS handshake timeout, try again
+						if apierrs.IsNotFound(err) ||
+							strings.Contains(err.Error(), "could not find default credentials") ||
+							strings.Contains(err.Error(), "TLS handshake timeout") {
```

Comment:
`apierrors.IsUnauthorized` and `apierrors.IsServerTimeout` would work better here.